build/webpack/webpack.config.base.js

```diff
@@ -75,9 +75,15 @@ module.exports = ({
 
     if (targetDeletesNodeGlobals) {
       plugins.push(new webpack.ProvidePlugin({
-        process: ['@electron/internal/common/webpack-provider', 'process'],
         global: ['@electron/internal/common/webpack-provider', '_global'],
-        Buffer: ['@electron/internal/common/webpack-provider', 'Buffer']
+        Buffer: ['@electron/internal/common/webpack-provider', 'Buffer'],
+        process: ['@electron/internal/common/webpack-provider', 'process']
+      }));
+    } else if (!alwaysHasNode) {
```

Comment:
I don't think `else if` is correct here, this should be a separate `if` statement